Error with Make: make: *** [Makefile:15: javascript] Error 2

@laemtl Thank you for the reply! I updated both JSON files but I am still getting the same errors.

`    ERROR in /var/www/loris/modules/electrophysiology_browser/jsx/react-series-data-viewer/src/series/components/IntervalSelect.tsx
    ./modules/electrophysiology_browser/jsx/react-series-data-viewer/src/series/components/IntervalSelect.tsx 98:11-15
    [tsl] ERROR in /var/www/loris/modules/electrophysiology_browser/jsx/react-series-data-viewer/src/series/components/IntervalSelect.tsx(98,12)
          TS2786: 'Rail' cannot be used as a JSX component.
      Its instance type 'Rail' is not a valid JSX element.
        The types returned by 'render()' are incompatible between these types.
          Type 'string | number | boolean | {} | Iterable<ReactNode> | ReactPortal | ReactElement<any, string | ((props: any) => ReactElement<any, any>) | (new (props: any) => Component<...>)>' is not assignable to type 'ReactNode'.
            Type '{}' is not assignable to type 'ReactNode'.
              Type '{}' is missing the following properties from type 'ReactPortal': key, children, type, props
    
    ERROR in /var/www/loris/modules/electrophysiology_browser/jsx/react-series-data-viewer/src/series/components/IntervalSelect.tsx
    ./modules/electrophysiology_browser/jsx/react-series-data-viewer/src/series/components/IntervalSelect.tsx 103:11-18
    [tsl] ERROR in /var/www/loris/modules/electrophysiology_browser/jsx/react-series-data-viewer/src/series/components/IntervalSelect.tsx(103,12)
          TS2786: 'Handles' cannot be used as a JSX component.
      Its instance type 'Handles' is not a valid JSX element.
        The types returned by 'render()' are incompatible between these types.
          Type 'string | number | boolean | {} | Iterable<ReactNode> | ReactPortal | ReactElement<any, string | ((props: any) => ReactElement<any, any>) | (new (props: any) => Component<...>)>' is not assignable to type 'ReactNode'.
            Type '{}' is not assignable to type 'ReactNode'.
              Type '{}' is not assignable to type 'ReactPortal'.
    
    ERROR in /var/www/loris/modules/electrophysiology_browser/jsx/react-series-data-viewer/src/series/components/IntervalSelect.tsx
    ./modules/electrophysiology_browser/jsx/react-series-data-viewer/src/series/components/IntervalSelect.tsx 117:11-16
    [tsl] ERROR in /var/www/loris/modules/electrophysiology_browser/jsx/react-series-data-viewer/src/series/components/IntervalSelect.tsx(117,12)
          TS2786: 'Ticks' cannot be used as a JSX component.
      Its instance type 'Ticks' is not a valid JSX element.
        The types returned by 'render()' are incompatible between these types.
          Type 'ReactElement<any, string | ((props: any) => ReactElement<any, any>) | (new (props: any) => Component<any, any, any>)>' is not assignable to type 'ReactNode'.
            Property 'children' is missing in type 'ReactElement<any, string | ((props: any) => ReactElement<any, any>) | (new (props: any) => Component<any, any, any>)>' but required in type 'ReactPortal'.
    
    ERROR in /var/www/loris/modules/electrophysiology_browser/jsx/react-series-data-viewer/src/eeglab/EEGLabSeriesProvider.tsx
    ./modules/electrophysiology_browser/jsx/react-series-data-viewer/src/eeglab/EEGLabSeriesProvider.tsx 165:7-15
    [tsl] ERROR in /var/www/loris/modules/electrophysiology_browser/jsx/react-series-data-viewer/src/eeglab/EEGLabSeriesProvider.tsx(165,8)
          TS2786: 'Provider' cannot be used as a JSX component.
      Its instance type 'Provider<AnyAction>' is not a valid JSX element.
        The types returned by 'render()' are incompatible between these types.
          Type 'React.ReactNode' is not assignable to type 'import("/var/www/loris/modules/electrophysiology_browser/jsx/react-series-data-viewer/node_modules/@visx/group/node_modules/@types/react/index").ReactNode'.
            Type '{}' is not assignable to type 'ReactNode'.
              Type '{}' is not assignable to type 'ReactPortal'.
make: *** [Makefile:15: javascript] Error 2

`

@laemtl Just an update, I did the remove commands and updated the JSON packages. 
This is the result. 

Also tried ``` npm cache clean --force ``` 

Output in terminal: 
```
lorisadmin@ubuntu-focal:/var/www/loris$ make
tools/gen-version.sh
npm install 
npm ERR! Invalid Version: ^7.1.16

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/lorisadmin/.npm/_logs/2022-09-28T15_06_53_305Z-debug-0.log
make: *** [Makefile:14: javascript] Error 1
```
 
Log file:
```
40 http fetch GET 200 https://registry.npmjs.org/@types%2freact-redux 55ms (cache hit)
41 timing idealTree Completed in 1693ms
42 timing command:install Completed in 1738ms
43 verbose stack TypeError: Invalid Version: ^7.1.16
43 verbose stack     at new SemVer (/usr/local/lib/node_modules/npm/node_modules/semver/classes/semver.js:38:13)
43 verbose stack     at compare (/usr/local/lib/node_modules/npm/node_modules/semver/functions/compare.js:3:32)
43 verbose stack     at Object.gte (/usr/local/lib/node_modules/npm/node_modules/semver/functions/gte.js:2:30)
43 verbose stack     at CanPlaceDep.checkCanPlaceCurrent (/usr/local/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/can-place-dep.js:173:51)
43 verbose stack     at CanPlaceDep.checkCanPlace (/usr/local/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/can-place-dep.js:157:27)
43 verbose stack     at new CanPlaceDep (/usr/local/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/can-place-dep.js:114:26)
43 verbose stack     at PlaceDep.place (/usr/local/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/place-dep.js:123:19)
43 verbose stack     at new PlaceDep (/usr/local/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/place-dep.js:73:10)
43 verbose stack     at /usr/local/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:944:31
43 verbose stack     at Array.map (<anonymous>)
```

Hey yes, it did help. Thank you! Made progress.

I was encountering 
``` 
lorisadmin@ubuntu-focal:/var/www/loris$ make
tools/gen-version.sh
npm install 
npm ERR! Invalid Version: ^17.0.1

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/lorisadmin/.npm/_logs/2022-09-29T15_03_22_245Z-debug-0.log
make: *** [Makefile:14: javascript] Error 1
``` 
I temporarily removed this issue by removing "^" from all instances of ^17.0.1.

Trying to solve another snaggle at the moment.
```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: react-series-data-viewer@1.0.0
npm ERR! Found: react@17.0.1
npm ERR! node_modules/react
npm ERR!   react@"17.0.1" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^15.0.0-0 || ^16.0.0-0" from @visx/group@1.17.1
npm ERR! node_modules/@visx/group
npm ERR!   @visx/group@"^1.0.0" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /home/lorisadmin/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/lorisadmin/.npm/_logs/2022-09-29T15_07_38_891Z-debug-0.log
npm ERR! code 1
npm ERR! path /var/www/loris
npm ERR! command failed
npm ERR! command sh -c -- cd modules/electrophysiology_browser/jsx/react-series-data-viewer && npm install

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/lorisadmin/.npm/_logs/2022-09-29T15_06_08_988Z-debug-0.log
make: *** [Makefile:14: javascript] Error 1
```

afterwards when I tried to troubleshoot electrophysiology_browser/jsx/react-series-data-viewer, this is my situation I am in currently. 
```
> react-series-data-viewer@1.0.0 preinstall
> npm install --package-lock-only --legacy-peer-deps --force --ignore-scripts && npx npm-force-resolutions

npm WARN using --force I sure hope you know what you are doing.
npm ERR! Linux 5.4.0-126-generic
npm ERR! argv "/usr/local/bin/node" "/var/www/loris/modules/electrophysiology_browser/jsx/react-series-data-viewer/node_modules/.bin/npm" "install" "--package-lock-only" "--legacy-peer-deps" "--force" "--ignore-scripts"
npm ERR! node v17.0.1
npm ERR! npm  v3.10.10

npm ERR! cb.apply is not a function
npm ERR! 
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     /var/www/loris/modules/electrophysiology_browser/jsx/react-series-data-viewer/npm-debug.log
npm ERR! code 1
npm ERR! path /var/www/loris/modules/electrophysiology_browser/jsx/react-series-data-viewer
npm ERR! command failed
npm ERR! command sh -c -- npm install --package-lock-only --legacy-peer-deps --force --ignore-scripts && npx npm-force-resolutions

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/lorisadmin/.npm/_logs/2022-09-29T15_30_13_844Z-debug-0.log

```
Will keep you updated.
Thanks for your help!

> Since your npm version is recent, there might be an easier way. Reset the changes made above in the main package.json: `> git checkout package.json`
> 
> and replace the content of modules/electrophysiology_browser/jsx/react-series-data-viewer/package.json:
> 
> ```
> {
>   "name": "react-series-data-viewer",
>   "homepage": "https://github.com/aces/react-series-data-viewer/",
>   "version": "1.0.0",
>   "description": "react-series-data-viewer React component",
>   "dependencies": {
>     "@visx/axis": "^1.4.0",
>     "@visx/group": "^1.0.0",
>     "@visx/responsive": "^1.3.0",
>     "@visx/shape": "^1.4.0",
>     "d3-3d": "0.0.10",
>     "d3-array": "^1.2.4",
>     "d3-dsv": "^1.0.10",
>     "d3-scale": "^2.1.2",
>     "d3-scale-chromatic": "^1.3.3",
>     "differenceequationsignal1d": "^0.1.1",
>     "gl-matrix": "^2.8.1",
>     "google-protobuf": "^3.6.1",
>     "ramda": "^0.25.0",
>     "react": "^16.12.0",
>     "react-compound-slider": "^3.3.1",
>     "react-dom": "^16.13.1",
>     "react-redux": "^7.2.1",
>     "redux": "^4.0.0",
>     "redux-actions": "^2.6.1",
>     "redux-logger": "^3.0.6",
>     "redux-observable": "^1.0.0",
>     "redux-thunk": "^2.3.0",
>     "resize-observer-polyfill": "^1.5.0",
>     "rxjs": "^6.6.3",
>     "tslib": "^1.9.3"
>   },
>   "devDependencies": {
>     "@types/react": "^16.12.0",
>     "@types/react-dom": "^16.9.9",
>     "@types/react-redux": "7.1.16"
>   },
>   "overrides": {
>     "@types/react": "^16.12.0"
>   },
>   "license": "MIT",
>   "repository": "https://github.com/aces/react-series-data-viewer"
> }
> ```
> 
> Run the following before running `make`:
> 
> ```
> rm -R /var/www/loris/node_modules/
> rm /var/www/loris/package-lock.json
> rm -R /var/www/loris/modules/electrophysiology_browser/jsx/react-series-data-viewer/node_modules/
> rm /var/www/loris/modules/electrophysiology_browser/jsx/react-series-data-viewer/package-lock.json
> ```

This is the result: 
``` Error - Can't read config.xml file. Webpack mode set to production.
Error: EACCES: permission denied, open '/var/www/loris/modules/mri_violations/js/columnFormatter.js'
make: *** [Makefile:15: javascript] Error 1```

Update on the situation:
    WARNING in ./node_modules/swagger-ui-dist/swagger-ui-bundle.js 2:993499
    Module parse failed: Unexpected token (2:993499)
    You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
    (Source code omitted for this binary file)
composer install --no-dev
> mkdir -p project/libraries
Installing dependencies from lock file
Verifying lock file contents can be installed on current platform.
Your lock file does not contain a compatible set of packages. Please run composer update.

  Problem 1
    - laminas/laminas-diactoros is locked to version 2.5.0 and an update of this package was not requested.
    - laminas/laminas-diactoros 2.5.0 requires php ^7.3 || ~8.0.0 -> your php version (8.1.11) does not satisfy that requirement.

make: *** [Makefile:4: all] Error 2
lorisadmin@ubuntu-focal:/var/www/loris$

--- a/modules/electrophysiology_browser/jsx/react-series-data-viewer/package.json
+++ b/modules/electrophysiology_browser/jsx/react-series-data-viewer/package.json
@@ -20,7 +20,7 @@
     "react": "^16.12.0",
     "react-compound-slider": "^3.3.1",
     "react-dom": "^16.13.1",
-    "react-redux": "^7.2.1",
+    "react-redux": "^8.0.2",
     "redux": "^4.0.0",
     "redux-actions": "^2.6.1",
     "redux-logger": "^3.0.6",

--- a/modules/electrophysiology_browser/jsx/react-series-data-viewer/src/series/components/AnnotationForm.tsx
+++ b/modules/electrophysiology_browser/jsx/react-series-data-viewer/src/series/components/AnnotationForm.tsx
@@ -1,6 +1,6 @@
 import React, {useEffect, useState} from 'react';
 import {Epoch as EpochType, RightPanel} from '../store/types';
-import {connect, DefaultRootState} from 'react-redux';
+import {connect} from 'react-redux';
 import {setTimeSelection} from '../store/state/timeSelection';
 import {setRightPanel} from '../store/state/rightPanel';
 import * as R from 'ramda';

--- a/modules/electrophysiology_browser/jsx/react-series-data-viewer/src/series/components/IntervalSelect.tsx
+++ b/modules/electrophysiology_browser/jsx/react-series-data-viewer/src/series/components/IntervalSelect.tsx
@@ -95,11 +95,13 @@ const IntervalSelect: FunctionComponent<CProps> = ({
             setIsDragging(false);
           }}
         >
+          {/* @ts-ignore */}
           <Rail>
             {({getRailProps}) => (
               <div style={railStyle} {...getRailProps()} />
             )}
           </Rail>
+          {/* @ts-ignore */}
           <Handles>
             {({handles, getHandleProps}) => (
               <div className="slider-handles">
@@ -114,6 +116,7 @@ const IntervalSelect: FunctionComponent<CProps> = ({
               </div>
             )}
           </Handles>
+          {/* @ts-ignore */}
           <Ticks count={10}>
             {({ticks}) => (
               <div

